src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -67,17 +97,46 @@ public virtual bool Check(VersionPayload version)
         }
     }
 
-    public delegate void NetworkPeerEventHandler(NetworkPeer peer);
-    public delegate void NetworkPeerEventMessageIncoming(NetworkPeer peer, IncomingMessage message);
-    public delegate void NetworkPeerStateEventHandler(NetworkPeer peer, NetworkPeerState oldState);
-
+    /// <summary>
+    /// Type of event handler that is triggered on network peer disconnection.
+    /// </summary>
+    /// <param name="peer">Network peer that was disconnected.</param>
+    public delegate void NetworkPeerDisconnectedEventHandler(NetworkPeer peer);
+
+    /// <summary>
+    /// Type of event handler that is triggered when a new message is received from a network peer.
+    /// </summary>
+    /// <param name="peer">Network peer from which the message was received.</param>
+    /// <param name="message">Message that was received.</param>
+    public delegate void NetworkPeerMessageReceivedEventHandler(NetworkPeer peer, IncomingMessage message);
+
+    /// <summary>
+    /// Type of event handler that is triggered when the network state of a peer was changed.
+    /// </summary>
+    /// <param name="peer">Network peer which network state was changed.</param>
+    /// <param name="oldState">Previous network state of the peer.</param>
+    public delegate void NetworkPeerStateChangedEventHandler(NetworkPeer peer, NetworkPeerState oldState);
+
+    /// <summary>Information about a message that the node sent to a peer.</summary>
     public class SentMessage
     {
+        /// <summary>Payload of the sent message.</summary>
         public Payload Payload;
+
+        /// <summary>
+        /// Completion of the send message task. 
+        /// </summary>
+        /// <remarks>
+        /// The result of the operation is set to <c>true</c> when the message is successfully sent to the peer.
+        /// It is never set to <c>false</c>, it can only fail if the peer is disconnected, in which case an exception is set on the completion.
```

Comment:
This is misleading no? it will default to false until task is complete.
But if any logic is using the task state itself then the bool value is not really required right?

As long as the Task is not completed is still running and an exception will set the Task state to failed with an error.

Reply:
so do you think the comment is not correct or that the design is suboptimal?

if the design, then that's ok, will be fixed later
if the comment, that's not ok, that should be fixed now

Reply:
nah forget it lets wait for your refactor (if any) then we can fix this comment.